### DISCUSSION TITLE
Link to dashboard by folder/name style "path"

### DISCUSSION THREAD

**Discussion by bobemoe** [grafana/grafana#59555] [Dashboards, Panels & Visualizations]
Is it possible to link to a dashboard by its name and folder name? I know the uuids are used for permanent URL's, which in most cases is desirable, but let me explain my use case:

We are monitoring hundreds of "devices".  We programmatically API generate several dashboards for each device. As well as some overview dashboards. This gives us a folder->dashboard structure like this. Dashboards in the Devices folder are named after their ID.

* General
  * Overview
  * Summary
* Device
  * Jim
  * Tony
  * Phil
  * ... 100's more auto generated

In Overview dashboard we have a query that returns graph of all devices (ID: Jim,Tony,Phil,...) and I want to add data links to the panel to link to their specific dashboard. The ID is contained in ${__series.name} but I dont know how to compose the link, we want something like `http://domain.tld/d/<folder name>/<dashboad name>` then we can do `http://domain.tld/d/Device/${__series.name}`

Is this existing functionality I've just missed? Or is there a better way to do this? I think there are some redirects available already like `/goto/` but they still need a dashboard uuid.

**Comment by tstibbs** [comment]
Given that this feature doesn't exist but Grafana is still well used, I would be interested to know how people are managing their dashboards such that they don't need to do this. Surely for a big organisation with many dashboards they would want links between them? Are they importing all the dashboards and then creating links manually...!? Or do they hand-crank everything in the Grafana UI and not bother with IaC?

Possible workaround: pre-assign the uids to dashboards. They don't _have_ to look like UUIDs, it seems to accept reasonably arbitrary strings. Because you then know which uid is assigned to which dashboard, you can then build links using the pre-known uids.

**Comment by mykael22000** [comment]
This would be pretty helpful, especially if we could also do it with a relative URL.

The basic model is an overview dashboard that links to detail dash board.  The catch is that we'd be distributing these dashboards to our customers to import into their Grafana instance, so I'm not sure if they'd even be created with the same guid (what's the scope of the guid?).  We also have no control over the sequence that they import the dashboards in, so it may not be possible to look it up when the other dashboard is imported.

Ideally, we want to specify a link from d/overview to d/detail (with some variables) and let Grafana resolve it as a local URL to locate the customers detail dashboard.

What we really don't want is for our customers to have to edit every overview dashboard to correct the guid for the detail dashboard.

If the guid is always preserved when our customers import the dashboard, we could hand edit all the guids in them, but that smells a lot like a quick and dirty hack with no guarantee that it'll keep working. 

We're looking at doing this for a couple of dozen pairs of dashboards.

**Comment by bobemoe** [comment]
My solution (workaround) to the specific use case in my OP was:

First to not use hand crafted dashboards with wildcards in the query for the overview dashboards, that way i don't need use ${__series.name} to get their id. Instead, move that logic and generation into the core app.

Since all of my Device dashboards were API generated by core app anyway, upon generation I store their returned UUID in core model/db. (or as @tstibbs suggests, pre-assign UUID)

That way the entire model (including the UUIDs) is available during the generation of all dashboards json, and the correct links can be formed during generation.

@mykael22000 can your core app generate or apply some search and replace to the json before it goes to client. Pretty sure the UUIDs are designed to persist across export/import.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
